src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -77,6 +78,60 @@ struct ExeSpaceUtils {
   static TeamPolicy get_team_policy_force_team_size (Int ni, Int team_size) {
     return TeamPolicy(ni, team_size);
   }
+
+  template <typename TeamMember, class Lambda, typename ValueType>
+  static KOKKOS_INLINE_FUNCTION
+  void parallel_reduce (const TeamMember& team,
+			const int& begin,
+			const int& end,
+                        const Lambda& lambda,
+                        const bool serialize,
```

Comment:
We definitely do not want this as a run-time argument. One suggestion is to make it a template parameter with default value based on EKAT's BFBness configuration.

Reply:
I've added it back as a template parameter and set based on `#ifdef EKAT_TEST_STRICT_FP`. But `EKAT_TEST_STRICT_FP` is only defined in the testing directory. Should I make it so that the definition is available everywhere?

Reply:
@tcclevenger I added a comment inside the code, but since it's discussed here: let's remove the default altogether, and have downstream app explicitly pass the last one. Apps can always do

```
#ifdef SOME_BFB_MACRO
constexpr bool SerialReduction = true;
#else
constexpr bool SerialReduction = false;
#endif
```